docs/paper/dev/api/roadmap.md

```diff
@@ -0,0 +1,93 @@
+---
+slug: /dev/roadmap
+---
+
+# Roadmap
+
+Paper offers a rich API that has a wide range of features that can help you unlock the full potential of your server. 
+However, in order to make room for new features and improvements, some of the older APIs will be phased out. This page 
+is intended to document any future API changes that are planned or possible deprecations that may be coming up. 
+
+## Future Plans
+
+
+### Interface ItemStacks
+
+When you create ItemStacks, you create an API representation of an ItemStack.
+However, there are also places where you can obtain an ItemStack that is backed by an NMS object instead. 
+This can lead to inconsistencies and unnecessary upkeep since we need to maintain our own implementation of ItemStack and
+support the NMS backed object using ugly methods due to it not being a plain interface.
+
+In the future, ItemStack will be converted to an interface that allows developers to use an underlying 
+ItemStack directly instead of going through the API implementation.
+
+#### Precautions
+- Avoid using the ItemStack constructor, instead use the static constructors
+    - `ItemStack.of(ItemType.DIAMOND_PICKAXE);`
+      - The ``ItemStack`` constructor will be removed at some point.
+- Avoid directly extending the ``ItemStack`` class.
+  - Custom implementations of this class are not supported
```

Comment:
This isn't possible without using the deprecated constructor as the ItemType constructor is private.

Reply:
Yes, but it's still possible and something we should note if people do this.